plugins/destination/s3/main.go

```diff
@@ -1,17 +1,28 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
+
 	"github.com/cloudquery/cloudquery/plugins/destination/s3/client"
-	"github.com/cloudquery/cloudquery/plugins/destination/s3/resources/plugin"
-	"github.com/cloudquery/plugin-sdk/v3/plugins/destination"
-	"github.com/cloudquery/plugin-sdk/v3/serve"
+	internalPlugin "github.com/cloudquery/cloudquery/plugins/destination/s3/resources/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/serve"
 )
 
 const (
 	sentryDSN = "https://c808d26943414033b2fb8bb5b5822ab9@o1396617.ingest.sentry.io/4504407917592576"
 )
 
 func main() {
-	p := destination.NewPlugin("s3", plugin.Version, client.New, destination.WithManagedWriter())
-	serve.Destination(p, serve.WithDestinationSentryDSN(sentryDSN))
+	p := plugin.NewPlugin("s3", internalPlugin.Version, client.New)
+	serveHelper(serve.Plugin(p, serve.WithPluginSentryDSN(sentryDSN), serve.WithDestinationV0V1Server()).Serve)
+}
+
+func serveHelper(f func(context.Context) error) {
```

Comment:
IMO this should just be in `main`

Reply:
it will probably removed, either moved to some sdk package or not accepted so I kept it easy to remove. :)